### PR TITLE
feat(demangle): Update Swift demangler to Swift 5.10

### PR DESCRIPTION
**Pull request by fantexi023**

### CODE REVIEW

File: symbolic-demangle/vendor/swift/1-arguments.patch [hunk 1]
```diff
@@ -1,45 +1,135 @@
...
 
-diff --git a/symbolic-demangle/vendor/swift/include/swift/Demangling/Demangle.h b/symbolic-demangle/vendor/swift/include/swift/Demangling/Demangle.h
-index db32dbd..f48e1c2 100644
---- a/symbolic-demangle/vendor/swift/include/swift/Demangling/Demangle.h
-+++ b/symbolic-demangle/vendor/swift/include/swift/Demangling/Demangle.h
-@@ -59,6 +59,7 @@ struct DemangleOptions {
-   bool ShortenArchetype = false;
-   bool ShowPrivateDiscriminators = true;
-   bool ShowFunctionArgumentTypes = true;
-+  bool ShowFunctionReturnType = true;
```
Comment (Swatinem) [inline symbolic-demangle/vendor/swift/1-arguments.patch:15]: You mentioned this in the changed `swiftdemangle.cpp` file. This option is indeed not part of upstream Swift, but we patched that in via this small patch here.

**Review by Swatinem — COMMENTED** [review]
Thank you for taking a look at this!
We have evidently not touched this code in quite a while, because updating it is quite cumbersome.

CI fails to compile with the updated code, but I can’t really make out what exactly the source of the problem is. Other than that, I added another comment related to a small patch that we applied to the swift code, which you should ideally carry forward.

File: symbolic-demangle/vendor/swift/1-arguments.patch (hunk 1 above)
Comment (fantexi023) [reply symbolic-demangle/vendor/swift/1-arguments.patch:15]: Thank you for your reply. 
I also saw that the CI process gave five errors, among which the problem of [Rust Test on macos-latest ](https://github.com/getsentry/symbolic/actions/runs/9989076444/job/27607390557#logs)should be that the SWIFT_STDLIB_HAS_TYPE_PRINTING did not take effect, which may be caused by the passing exception of CMakeLists.txt. I cancelled the restriction of SWIFT_STDLIB_HAS_TYPE_PRINTING this macro locally. The other four problems correspond to the same error, but I did not encounter this error on my own Mac.
Lastly, regarding ShowFunctionReturnType I still don't understand, how do you work through patches? Isn't the final demangle still implemented through the official Swift interface?

Comment (fantexi023) [reply symbolic-demangle/vendor/swift/1-arguments.patch:15]: > You mentioned this in the changed `swiftdemangle.cpp` file. This option is indeed not part of upstream Swift, but we patched that in via this small patch here.

yeah, I do understand your explanation for ShowFunctionReturnType. TXS!

**Comment by loewenheim** [comment]
@fantexi023 Are you still working on this?

**Comment by loewenheim** [comment]
Closing this for now, we might pick it up again later.

**Comment by Dav1dde** [comment]
Followed up on in #885 with an update to 6.0.3

**Pull request closed without merging**